### PR TITLE
[ci] Update Chrome install script and version.

### PR DESCRIPTION
**Pull request by ditman**
This brings the version of Chrome that we use for integration web tests in line with the one used by flutter/flutter and flutter/engine.

No Issue (that I could find); at first I thought this would help with a test that I'm writing for an unrelated issue, but then I thought that the improvements to the script might be interesting (we might want to rollback the chromium build_id to what was there before?)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: script/install_chromium.sh [hunk 1]
```diff
@@ -10,24 +10,35 @@ readonly TARGET_DIR=$1
...
+
+# The correct ChromeDriver is distributed alongside the chromium build above, as
+# `chromedriver_linux64.zip`, so no need to hardcode any extra info about it.
+readonly DOWNLOAD_ROOT="https://www.googleapis.com/download/storage/v1/b/chromium-browser-snapshots/o/Linux_x64%2F${CHROMIUM_BUILD}%2F"
 
 # Install Chromium.
-mkdir "$TARGET_DIR"
-wget --no-verbose "https://www.googleapis.com/download/storage/v1/b/chromium-browser-snapshots/o/Linux_x64%2F${CHROMIUM_BUILD}%2Fchrome-linux.zip?alt=media" -O "$TARGET_DIR"/chromium.zip
-unzip "$TARGET_DIR"/chromium.zip -d "$TARGET_DIR"/
+mkdir $TARGET_DIR
```
Comment (stuartmorgan) [inline script/install_chromium.sh:22]: Why did you remove the quotes here?

File: script/install_chromium.sh [hunk 2]
```diff
@@ -10,24 +10,35 @@ readonly TARGET_DIR=$1
...
+readonly CHROMIUM_ZIP_FILE="$TARGET_DIR/chromium.zip"
+wget --no-verbose "${DOWNLOAD_ROOT}chrome-linux.zip?alt=media" -O $CHROMIUM_ZIP_FILE
+unzip -q $CHROMIUM_ZIP_FILE -d "$TARGET_DIR/"
```
Comment (stuartmorgan) [inline script/install_chromium.sh:25]: Why no quotes on the source file?

File: script/install_chromium.sh [hunk 3]
```diff
@@ -10,24 +10,35 @@ readonly TARGET_DIR=$1
...
 
 # Install ChromeDriver.
 readonly DRIVER_ZIP_FILE="$TARGET_DIR/chromedriver.zip"
-wget --no-verbose "https://chromedriver.storage.googleapis.com/$CHROME_DRIVER_VERSION/chromedriver_linux64.zip" -O "$DRIVER_ZIP_FILE"
-unzip "$DRIVER_ZIP_FILE" -d "$TARGET_DIR/chromedriver"
+wget --no-verbose "${DOWNLOAD_ROOT}chromedriver_linux64.zip?alt=media" -O $DRIVER_ZIP_FILE
+unzip -q $DRIVER_ZIP_FILE -d "$TARGET_DIR/"
```
Comment (stuartmorgan) [inline script/install_chromium.sh:30]: Same question

File: script/install_chromium.sh [hunk 4]
```diff
@@ -10,24 +10,35 @@ readonly TARGET_DIR=$1
...
-# The ChromeDriver version corresponding to the build above. See
-# https://chromedriver.chromium.org/downloads
-# for versions mappings when updating Chromium.
-readonly CHROME_DRIVER_VERSION=84.0.4147.30
+#
+# Check: https://github.com/flutter/engine/blob/master/lib/web_ui/dev/browser_lock.yaml
+readonly CHROMIUM_BUILD=929514
+
+# The correct ChromeDriver is distributed alongside the chromium build above, as
+# `chromedriver_linux64.zip`, so no need to hardcode any extra info about it.
```
Comment (stuartmorgan) [inline script/install_chromium.sh:18]: Do we know if this will be true in general going forward?

File: script/install_chromium.sh [hunk 5]
```diff
@@ -10,24 +10,35 @@ readonly TARGET_DIR=$1
...
+
+export CHROME_EXECUTABLE="$TARGET_DIR/chrome-linux/chrome"
 
 # Echo info at the end for ease of debugging.
-export CHROME_EXECUTABLE="$TARGET_DIR"/chrome-linux/chrome
+set +x
+echo
+readonly CHROMEDRIVER_EXECUTABLE="$TARGET_DIR/chromedriver/chromedriver"
 echo $CHROME_EXECUTABLE
 $CHROME_EXECUTABLE --version
```
Comment (stuartmorgan) [inline script/install_chromium.sh:41]: This should be quoted too actually (along with the new call below).

File: script/install_chromium.sh (hunk 1 above)
Comment (ditman) [reply script/install_chromium.sh:22]: Because I had no idea what I was doing, but I've done some reading now :P

File: script/install_chromium.sh (hunk 4 above)
Comment (ditman) [reply script/install_chromium.sh:18]: This has been true since February 2019:

* https://commondatastorage.googleapis.com/chromium-browser-snapshots/index.html?prefix=Linux_x64/634372/ 

But I don't have any guarantees that this is going to continue to be true in the future (but it's very convenient as is!). Let me ask in YAQS, see if anybody responds.

File: script/install_chromium.sh (hunk 1 above)
Comment (ditman) [reply script/install_chromium.sh:22]: https://tldp.org/LDP/abs/html/quotingvar.html

File: script/install_chromium.sh (hunk 4 above)
Comment (stuartmorgan) [reply script/install_chromium.sh:18]: > This has been true since February 2019

That sounds like the kind of "temporary" that's not likely to change a lot, so it seems fine to depend on. It's not like it's hard to go back to the old approach if it's changed without something better (and it sounds like it would probably remain this way until there's something better).

**Review by stuartmorgan — APPROVED** [review]
LGTM

File: script/install_chromium.sh (hunk 4 above)
Comment (ditman) [reply script/install_chromium.sh:18]: Yes, if this assumption changes, we'd also need to change the CIPD roller script in the engine repo (but that thing only runs seldomly, this script would break much earlier)

**Comment by ditman** [comment]
video player seems to be legitimately failing:

```
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════
The following PlatformException was thrown running a test:
PlatformException(VideoError, Failed to load video: The requested
URL was not found on this server., null, null)
```

**Comment by stuartmorgan** [comment]
Transient server failure maybe? I really need to finish getting https://github.com/flutter/plugins/pull/4987 working to eliminate that URL.

**Comment by ditman** [comment]
> Transient server failure maybe?

Looks like either a transient failure, or that we're getting throttled/rate limited. After a rerun, a different package failed (both tests use the same m3u8 URL, though!):

First:
```
The following packages had errors:
  video_player_avfoundation:
```

Then:

```
The following packages had errors:
  video_player/video_player:
```

> I extracted the video_player part of that PR into https://github.com/flutter/plugins/pull/5103 since they are particular problematic (as demonstrated here).

Thanks for prioritizing that fix, I'll take a look!

**Comment by stuartmorgan** [comment]
That's landed, so hopefully merging `main` in will fix this test.

**Comment by ditman** [comment]
Looking much better in the current run (All ios test seem to have passed now). _(Knock on wood.)_

**Comment by ditman** [comment]
(No tests for this .sh file yet, I guess it could get migrated as a dart tool?)

**Comment by stuartmorgan** [comment]
> (No tests for this .sh file yet, I guess it could get migrated as a dart tool?)

Potentially, but it's so simple that I tend to categorize it as "it is a test"; there's no way for it to fail without the tests that need it failing. (Unlike most of the repo tooling, where silently not failing when it should is a very real danger that's happened a bunch of times in the past.) We could potentially even teach the bot to ignore that specific file.

**Comment by ditman** [comment]
> I tend to categorize it as "it is a test"; there's no way for it to fail without the tests that need it failing.

Yeah, that is a very good point!
